tests/reference_images.rs

```diff
@@ -95,49 +97,135 @@ fn check_references() {
             let mut path: Vec<_> = path.components().collect();
             (path.pop().unwrap(), path.pop().unwrap())
         };
+
+        let mut filename_parts = filename
+            .as_os_str()
+            .to_str()
+            .unwrap()
+            .split('.')
+            .collect::<Vec<_>>();
+
+        // Ignore the file extension
+        filename_parts.pop().unwrap();
+
+        // The penultimate part of `filename_parts` represents the metadata.
+        let meta_str = filename_parts.pop().unwrap();
+        let mut meta = meta_str.split('_').collect::<Vec<_>>();
```

Comment:
Can this, together with the `if` conditions below, be a seperate method that returns a `Option<TestKind>` where `TestKind` is an enum with two variants for the two cases.

The whole parsing of test metadata creates a lot of clutter before the actual test and code flow it isn't really obvious which assertions are part of the test itself.